linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -1120,7 +1121,7 @@ private[emitter] final class ClassEmitter(jsGen: JSGen) {
             val decodedName = className.nameString.stripSuffix("$")
             val msg = s"Initializer of $decodedName called before completion " +
               "of its super constructor"
-            val obj = js.New(encodeClassVar(UndefinedBehaviorErrorClass), Nil)
+            val obj = genScalaClassNew(UndefinedBehaviorErrorClass, NoArgConstructorName)
             val ctor = obj DOT js.Ident(genName(StringArgConstructorName))
             js.Throw(js.Apply(ctor, js.StringLiteral(msg) :: Nil))
```

Comment:
Consider moving this logic in a helper `$throwModuleInitError(decodedName)` defined once and for all by the core JS lib (only when `semantics.moduleInit` is Fatal). Then here we would just call `genCallHelper("throwModuleInitError", js.StringLiteral(decodedName))`.

This way, we do not need to require a global knowledge for `genModuleAccessor`, and hence we do not need to remove its caching in the emitter. It's better for code size too.

This change could be done in a separate commit that comes before this commit.